### DISCUSSION TITLE
How to run Directus permanently on Debian?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#16428] [Archive]
From PuTTy, `npx directus start` works.
But how to keep it running when I close PuTTy?

Should I use PM2? nohut?
And which script is the `server.js`?

I tried 
nohup node /var/www/vhosts/mydomain.com/app/node_modules/.bin/directus > nohup.out 2> nohup.err < /dev/null &
with no success.
The answer is `[1] 200445`. Directus server is not running.

**Comment by domdus** [comment]
pm2 works great.

- npm install pm2 -g
- Go to directus app dir where package.json is located
- pm2 start npm --name "any_name_for_your_app" --run start
- pm2 save    (persist process to survive server reboot)

**Reply by Pachat** [reply]
Thank you for this solution I use like this

- `npm install pm2 -g`
- `nano directus.sh` (in directus app dir where package.json is located)
`npx directus start`
- `pm2 start directus.sh`
- `pm2 startup` (gives a command to paste to register the startup)
- `pm2 save`

- sudo reboot
and Directus is alive :)

**Comment by hugopvl** [comment]
or if you want, you can use contab -e  -->  @reboot sh -c 'cd /root/yourappname && npx directus start'

**Reply by Pachat** [reply]
Very simple indeed
